onadata/apps/api/viewsets/data_viewset.py

```diff
@@ -312,6 +312,7 @@ def destroy(self, request, *args, **kwargs):
                     self.object.instances.filter(
                         deleted_at__isnull=True).update(
                             deleted_at=timezone.now(),
+                            date_modified=timezone.now(),
                             deleted_by=request.user)
```

Comment:
@DavisRayM date_modified is supposed to be auto-filled by the database

Reply:
@denniswambua `date_modified` is [auto filled by django](https://docs.djangoproject.com/en/3.2/ref/models/fields/#django.db.models.DateField.auto_now) when you call the `save()` function. `update()` usually doesn't call the `save()` function